3574/1/read_file.rb

```diff
@@ -0,0 +1,23 @@
+class ReadFile
+  def call(for_file)
+    processing_file(for_file)
+  end
+
+  private
+
+  def processing_file(for_file)
+    names = []
+    read_file(for_file, names)
+    names.shift
+    names
+  end
+
+  def read_file(for_file, names)
```

Comment:
you are reading yaml file so please use YAML module instead. it will significantly shorten your code